kubeconfig/src/Kubernetes/KubeConfig.hs

```diff
@@ -89,14 +90,34 @@ data NamedEntity a (typeKey :: Symbol) = NamedEntity
 
 instance (FromJSON a, Typeable a, KnownSymbol s) =>
          FromJSON (NamedEntity a s) where
-  parseJSON = withObject ("Named" <> (show $ typeOf (undefined :: a))) $ \v ->
+  parseJSON = withObject ("Named" <> (show $ typeOf (Proxy :: Proxy a))) $ \v ->
     NamedEntity <$> v .: "name" <*> v .: T.pack (symbolVal (Proxy :: Proxy s))
 
 instance (ToJSON a, KnownSymbol s) =>
          ToJSON (NamedEntity a s) where
   toJSON (NamedEntity {..}) = object
       ["name" .= toJSON name, T.pack (symbolVal (Proxy :: Proxy s)) .= toJSON entity]
 
+newtype Spec = Spec {containers :: [Container]}
+  deriving (Eq, Generic, Show, Typeable)
+
+data Container = Container {
```

Comment:
You don't need to define a Container type. Use V1Container in https://raw.githubusercontent.com/kubernetes-client/haskell/master/kubernetes/lib/Kubernetes/Model.hs.